portalnet/src/overlay/service.rs

```diff
@@ -2461,6 +2589,82 @@ fn pop_while_ssz_bytes_len_gt(enrs: &mut Vec<SszEnr>, max_size: usize) {
     }
 }
 
+/// References to `OverlayService` components required for processing
+/// a utp stream. This is basically a utility struct to avoid passing
+/// around a large number of individual references.
+struct UtpProcessing<TValidator, TStore, TContentKey>
```

Comment:
I'm not sure about this name, but it's quite useful to have this structure to pass around vs. individually handling its components. It seems like there are opportunities to further refactor the `OverlayService` using this datatype, but I felt best to leave that for a future pr to avoid irrelevant bloat to this pr.